src/datasets/summarizations/libs/trend.ts

```diff
@@ -141,3 +148,69 @@ export function exponentialMovingAverage(points: TimeSeriesPoint[], alpha = 0.3)
   }));
   return smoothedPoints;
 }
+
+export function centeredMovingAverage(points: TimeSeriesPoint[], k: number): TimeSeriesPoint[] {
+  const L = points.length;
+  const smoothedPoints: TimeSeriesPoint[] = [];
+  for (let i = 0; i < points.length; i++) {
+    const lSumPoints = points.slice(Math.max(0, i - k), Math.min(L, i + k));
+    const rSumPoints = points.slice(Math.max(0, i - k + 1), Math.min(L, i + k + 1));
+
+    const lSum = sum(lSumPoints.map(({ y }) => y));
+    const rSum = sum(rSumPoints.map(({ y }) => y));
+
+    const smoothedY = 0.5 * (lSum / lSumPoints.length + rSum / rSumPoints.length);
+    smoothedPoints.push({
+      x: points[i].x,
+      y: smoothedY,
+    });
+  }
+  return smoothedPoints;
+}
+
+export type GroupIdentifier = string | number;
+export interface DecompositionResult {

```

Comment:
Please add description to interface and properties. Are all the points separated into one of these categories? Can a point be in more than one category? What do these categories mean?

Reply:
I added JSDoc comments for `additiveDecomposite` and explained the relationships between the inputs and outputs there.